src/Disks/MetadataStorageFromBackupFile.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include <unordered_map>
+#include <vector>
+#include <Disks/ObjectStorages/IMetadataStorage.h>
+#include <Disks/IDisk.h>
+
+
+namespace DB
+{
+
+class MetadataStorageFromBackupFile final : public IMetadataStorage
+{
+private:
+    friend class MetadataStorageFromBackupFileTransaction;
+
+    struct MetadataStorageFromBackupFilePseudoFileSystemNode
+    {
+    public:
+        String name;
+        String path;
+        uint64_t file_size = 0;
+        bool is_file = false;
+        bool is_directory = false;
+        std::vector<String> children = {};
+    };
+    std::unordered_map<String /* path */, MetadataStorageFromBackupFilePseudoFileSystemNode> nodes;
+
+public:
+    explicit MetadataStorageFromBackupFile(const String & path_to_backup_file);
```

Comment:
I do not see an object of this class being created anywhere, or I missed it somewhere?